### ISSUE TITLE
SC2155 not detected in parameter-expansion default-value

### ISSUE THREAD

**Issue by leagris** [koalaman/shellcheck#2456]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2155
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

fn() {
        printf 'Hello world!\n'
}

# SC2155 unexpectedly not triggered here
declare a=${1:-$(fn)}

# SC2155 triggered here as expected
declare b=$(fn)

printf '%s\n' "$a" "$b"
```

#### Here's what shellcheck currently says:
```
Line 11	SC2155: Declare and assign separately to avoid masking return values.
```
#### Here's what I wanted or expected to see:

I'd expect Shellcheck to detect command substitutions within parameter expansion default value `word`.

In the code snippet above it should also report:
```
Line 8	SC2155: Declare and assign separately to avoid masking return values.
```

---

##### Bash parameter expansion Default-Values:

       ${parameter:-word}
              Use Default Values.  If parameter is unset or null,  the  expan‐
              sion  of word is substituted.  Otherwise, the value of parameter
              is substituted.
       ${parameter:=word}
              Assign Default Values.  If parameter is unset or null,  the  ex‐
              pansion  of word is assigned to parameter.  The value of parame‐
              ter is then substituted.  Positional parameters and special  pa‐
              rameters may not be assigned to in this way.
       ${parameter:?word}
              Display  Error if Null or Unset.  If parameter is null or unset,
              the expansion of word (or a message to that effect  if  word  is
              not  present) is written to the standard error and the shell, if
              it is not interactive, exits.  Otherwise, the value of parameter
              is substituted.
       ${parameter:+word}
              Use  Alternate Value.  If parameter is null or unset, nothing is
              substituted, otherwise the expansion of word is substituted.
